File: JAVA/S1/L1/OOPS/encapsulation.java
class bankaccount{
    private String AccountHolder;
    private double balance;
    bankaccount(String AccountHolder,double balance)
    {
        this.AccountHolder=AccountHolder;
        this.balance=balance;
    }

    public String GetAccountHolder()
    {
        return AccountHolder;  
    }
    public void SetAccountHolder(String AccountHolder)
    {
        this.AccountHolder=AccountHolder;
    }
    public void deposit(double amount)
    {
        if(amount>0)
        {
            balance=balance+amount;
        }
    }
    public void withdraw(double amount)
    {
        if(amount>0 &&amount<=balance)
        {
            balance=balance-amount;

        }
    }
    public double getbalance()
    {
        return balance;
    }

}
public class encapsulation
{
    public static void main(String[] args) {
        {
            bankaccount acc=new bankaccount("abhi",2000);
            acc.deposit(78);
            acc.withdraw(262);
            System.out.println(acc.getbalance());
            acc.SetAccountHolder("alex");
            System.out.println(acc.GetAccountHolder());
        }
    }
}

File: CPP/oops/encapsulation.cpp
/*
Encapsulation is the process of hiding the internal details (data) of a class and providing a controlled way to access or modify it.

Keeps data safe from outside interference or misuse.

Achieved using access specifiers: private, protected, and public.
*/
#include<iostream>
using namespace std;
class BankAccount{
    private:
    double balance;
    public:
    BankAccount(double initialbalance)
    {
        if(initialbalance>=0)
        {
            balance=initialbalance;

        }
        else{
            balance=0;
        }
    }
    void deposit(double amount)
    {
        if(amount>=0)
        {
            balance+=amount;
        }
        else{
            cout<<"money can not be deposited";
        }
    }
    double getbalance()
    {
        return balance;
    }
};
int main()
{
    BankAccount ac(1000);
    ac.deposit(200);
    cout<<ac.getbalance();
    return 0;
}